### ISSUE TITLE
fusion.py — correctness bug, marginalized compatibility guard, and test coverage

### ISSUE THREAD

**Issue by stepanyanhayk** [McDermottHealthAI/MedRAP#55]
## Bug fix

**`ReplaceFusion.fuse` wrong squeeze condition**
`fusion.py:104`:
```python
elif s_doc == 1:
    fused_state = rm.squeeze(1).squeeze(2)
```
When `R > 1` and `S_doc = 1`, neither dim 1 nor dim 2 is size 1 so both squeezes are no-ops, returning a 5D tensor instead of 3D. Fix: `elif r == 1 and s_doc == 1:`.

## Marginalized retrieval compatibility

**No guard ensuring `marginalized_retrieval=True` is only used with `ReplaceFusion`**
`model.py` checks `fused_state.ndim == 3` but not whether dim 1 truly represents K retrieved documents. `PassthroughFusion` and `CrossAttentionFusion` both return `(B, S_ehr, D)` which passes the ndim check — the model silently marginalizes over EHR sequence positions instead of documents, producing wrong `per_doc_logits`.

Only `ReplaceFusion` (with a per-doc retrieval encoder) correctly produces `(B, K, D)`. Add an explicit check in `model.py` raising a clear error if `marginalized_retrieval=True` is combined with an incompatible fusion, or at minimum document this constraint prominently in `RetrievalAugmentedModel`.

## Design note

**`ConcatFusion` only supports `S=1` patient states**
The validation enforces `patient_state.shape[1] == 1`, making it incompatible with sequence encoders like `TimeDeltaRoPEPatientEncoder`. With `TabularEncoder` being removed (see #51), the only compatible encoder left is `MEDSCodeEncoder` (a test scaffold). Decide whether to generalise `ConcatFusion` to arbitrary sequence lengths or remove it.

**`PassthroughFusion` wastes compute when used as ablation**
The retriever and retrieval encoder still run full forward passes even though their output is discarded. In an ablation run these passes consume memory and compute for nothing. Consider skipping them at the pipeline level when `PassthroughFusion` is active.

## Tests

**Add `R > 1, S_doc = 1` doctest to `ReplaceFusion.fuse`**
This is the buggy branch — a doctest covering this shape serves as a regression test once the fix is in.

**Add `R > 1` doctest to `CrossAttentionFusion.fuse`**
All existing examples use `R = 1`. A test with `R = 2` verifies the `r * k * s_doc` flattening is correct across multiple retrieval rounds.
